force-app/main/default/classes/StringGenerator.cls

```diff
@@ -0,0 +1,30 @@
+public with sharing class StringGenerator {
+
+    private List<String> myList = new List<String>();
+
+    public StringGenerator( List<String> myListByUser )
+    {
+        myList = myListByUser;
+    }
+    
+    public List<String> returnRandomString()
+    {
+        List<Integer> randomIndexList = new List<integer>();
+        
+        List<String> randomStringList = new List<String>();
+        
+        Integer randomIndex;
```

Comment:
the scope of randomIndex is within the while loop. 
It should be defined in the loop